packages/ember-routing/lib/system/route.js

```diff
@@ -811,18 +811,26 @@ var Route = EmberObject.extend(ActionHandler, Evented, {
 
       var handlerInfos = transition.state.handlerInfos;
       var router = this.router;
-      var qpMeta = router._queryParamsFor(handlerInfos[handlerInfos.length - 1].name);
+      var qpMeta;
       var changes = router._qpUpdates;
       var replaceUrl;
 
-      stashParamNames(router, handlerInfos);
+      if (!handlerInfos.length) {
```

Comment:
So we don't want to `stashParamNames` if there are no `handlerInfos`?  This is to ensure that `#setup` has been called first, right?  If so, is there a better way to check this (or add a comment that explains why)?